npm/test/unit/utils/ldapUtils.test.js

```
@@ -0,0 +1,304 @@
+      const group = {
+        name: 'ldapgroup',
+        gid_number: 4000,
+        members: ['uid=user1,ou=users,dc=example,dc=com', 'uid=user2,ou=users,dc=example,dc=com']
+      };
+      
+      const entry = createLdapGroupEntry(group, baseDN);
+      
+      expect(entry.attributes.member).toEqual([
+        'uid=user1,ou=users,dc=example,dc=com',
+        'uid=user2,ou=users,dc=example,dc=com'
+      ]);
+    });
+    
+    test('should work with all test fixture groups', () => {
+      testGroups.forEach(group => {
+        const entry = createLdapGroupEntry({
+          name: group.cn,
+          gid_number: group.gidNumber,
+          memberUids: group.memberUids
+        }, baseDN);
+        
+        expect(entry.dn).toContain(group.cn);
+        expect(entry.attributes.cn).toBe(group.cn);
+        expect(entry.attributes.gidNumber).toBe(group.gidNumber);
+      });
+    });
+  });
+  
+  describe('extractDomainFromBaseDn', () => {
+    
+    test('should extract domain from standard base DN', () => {
+      const domain = extractDomainFromBaseDn('dc=example,dc=com');
+      expect(domain).toBe('example.com');
+    });
+    
+    test('should handle multi-level domains', () => {
+      const domain = extractDomainFromBaseDn('dc=mail,dc=example,dc=com');
+      expect(domain).toBe('mail.example.com');
+    });
+    
+    test('should handle single-level domain', () => {
+      const domain = extractDomainFromBaseDn('dc=localhost');
+      expect(domain).toBe('localhost');
+    });
+    
+    test('should handle base DN with OU components', () => {
+      const domain = extractDomainFromBaseDn('ou=users,dc=company,dc=org');
+      expect(domain).toBe('company.org');
+    });
+    
+    test('should return localhost for invalid base DN', () => {
+      const domain = extractDomainFromBaseDn('invalid-dn');
+      expect(domain).toBe('localhost');
+    });
+    
+    test('should return localhost for empty base DN', () => {
```

Empty base would mean empty domain to me. Localhost seems like the wrong default. Or we should throw an error that an empty DN was provided.